chat-widget/src/common/utils.ts

```diff
@@ -285,4 +286,34 @@ export const getWidgetCacheId = (orgId: string, widgetId: string): string => {
 
 export const getWidgetEndChatEventName = (orgId: string, widgetId: string): string => {
     return `${BroadcastEvent.ChatEnded}_${orgId}_${widgetId}`;
+};
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export const getStateFromCache = (orgId: string, widgetId: string): any => {
+    // Getting updated state from cache
+    if (DataStoreManager.clientDataStore) {
+        const widgetStateEventName = getWidgetCacheId(orgId, widgetId);
+        const widgetStateFromCache = DataStoreManager.clientDataStore?.getData(widgetStateEventName, "localStorage");
+        const persistedState = widgetStateFromCache ? JSON.parse(widgetStateFromCache) : undefined;
+        return persistedState;
+    } else {
+        return null;
+    }
+};
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export const isUndefinedOrEmpty = (object: any) => {
+    if (object) {
+        if (Object.keys(object).length === 0) {
+            return true;
+        }
+        return false;
+    } else {
```

Comment:
Nit: No need  of else block

Reply:
its needed for undefined case

Reply:
You can simply return true. No need of else